mediator/med_map_mod.F90

```diff
@@ -931,66 +951,131 @@ end subroutine med_map_FB_Regrid_Norm_All
+    integer               , intent(out)          :: rc
+    ! ----------------------------------------------
+
+    ! local
+    type(ESMF_Field)       :: field1, field2
+    character(CS)          :: lfldname
+    character(len=*),parameter :: subname='(med_map_FB_Field_Regrid)'
+    ! ----------------------------------------------
+
+    if (dbug_flag > 10) then
+       call ESMF_LogWrite(trim(subname)//": start", ESMF_LOGMSG_INFO)
+    endif
+
+    call t_startf(subname)
+    rc = ESMF_SUCCESS
+
+    lfldname=trim(fldin)//'->'//trim(fldout)
+
+    call ESMF_LogWrite(trim(subname)//": called", ESMF_LOGMSG_INFO)
+
+    if (FB_FldChk(FBin , trim(fldin) , rc=rc) .and. &
+        FB_FldChk(FBout, trim(fldout), rc=rc)) then
+
+       call FB_GetFieldByName(FBin, trim(fldin), field1, rc=rc)
+       if (chkerr(rc,__LINE__,u_FILE_u)) return
+
+       call FB_GetFieldByName(FBout, trim(fldout), field2, rc=rc)
+       if (chkerr(rc,__LINE__,u_FILE_u)) return
+
+       call med_map_Field_Regrid(field1, field2, RouteHandles, mapindex, subname//trim(lfldname), rc=rc)
+       if (chkerr(rc,__LINE__,u_FILE_u)) return
+    else
+       call ESMF_LogWrite(trim(subname)//" field not found: "//&
+            trim(fldin)//","//trim(fldout), ESMF_LOGMSG_INFO)
+    endif
+
+    if (dbug_flag > 10) then
+       call ESMF_LogWrite(trim(subname)//": done", ESMF_LOGMSG_INFO)
+    endif
+    call t_stopf(subname)
+
+   end subroutine med_map_FB_Field_Regrid
+
+  !================================================================================
+
+  subroutine med_map_Field_Regrid (srcfield, dstfield, RouteHandles, mapindex, fldname, rc)
 
     !---------------------------------------------------
     ! map the source field to the destination field
     !---------------------------------------------------
 
     use ESMF , only : ESMF_LogWrite, ESMF_LOGMSG_INFO, ESMF_SUCCESS
     use ESMF , only : ESMF_LOGMSG_ERROR, ESMF_FAILURE, ESMF_MAXSTR
-    use ESMF , only : ESMF_FieldBundle, ESMF_FieldBundleGet
-    use ESMF , only : ESMF_Field, ESMF_FieldGet, ESMF_FieldRegrid
+    use ESMF , only : ESMF_Field, ESMF_FieldRegrid
     use ESMF , only : ESMF_TERMORDER_SRCSEQ, ESMF_Region_Flag, ESMF_REGION_TOTAL
     use ESMF , only : ESMF_REGION_SELECT
-    use ESMF , only : ESMF_RouteHandle, ESMF_RouteHandleIsCreated
+    use ESMF , only : ESMF_RouteHandle
 
     ! input/output variables
-    character(len=*)       , intent(in)    :: fldname
     type(ESMF_Field)       , intent(in)    :: srcfield
     type(ESMF_Field)       , intent(inout) :: dstfield
     type(ESMF_RouteHandle) , intent(inout) :: RouteHandles(:)
     integer                , intent(in)    :: mapindex
+    character(len=*)       , intent(in), optional :: fldname
     integer                , intent(out)   :: rc
 
     ! local variables
     logical :: checkflag = .false.
+    character(len=CS) :: lfldname
+    character(len=*), parameter    :: subname='(module_MED_Map:med_map_Field_Regrid)'
```

Comment:
Of course this will be a part of separate PR but it would be nice to have a standard in surname naming. For example we are using two level in here but "med_map_FB_Field_Regrid" has only one level.